snuba/request/validation.py

```diff
@@ -1,25 +1,19 @@
-import jsonschema
 import sentry_sdk
-from werkzeug.exceptions import BadRequest
 
 from snuba.datasets.dataset import Dataset
 from snuba.request import Request
 from snuba.request.schema import RequestSchema
 from snuba.utils.metrics.timer import Timer
 
 
-def validate_request_content(
+def build_request(
     body, schema: RequestSchema, timer: Timer, dataset: Dataset, referrer: str
 ) -> Request:
-    with sentry_sdk.start_span(
-        description="validate_request_content", op="validate"
-    ) as span:
-        try:
-            request = schema.validate(body, dataset, referrer)
-            span.set_data("snuba_query", request.body)
-        except jsonschema.ValidationError as error:
-            raise BadRequest(str(error)) from error
```

Comment:
I think this was not the right exception. BadRequest is coupled with HTTP400. This code does not need to be coupled with the transport being HTTP. It just need to know the payload is json inn input and the output should be a Request. 
We actually have a case where this code is invoked outside of the context of an HTTP request (subscription worker).